src/components/playback/playback-control.tsx

```diff
@@ -42,26 +41,17 @@ export const PlaybackControlComponent: React.FC<IProps> = observer((props: IProp
   // by the history stuff, but it is being used to trigger document saves to Firebase
   // I think.  In some sense this is like a hash of the document content.
 
-  const {numHistoryEventsApplied} = treeManager;
+  const {numHistoryEventsApplied, currentHistoryEntry} = treeManager;
   // numHistoryEventsApplied can be 0 or undefined, the event is undefined in both cases
-  const currentHistoryEvent = !numHistoryEventsApplied
-                                ? undefined
-                                : treeManager.getHistoryEntry(numHistoryEventsApplied - 1);
-  const sliderValue = numHistoryEventsApplied === undefined ? 0 : numHistoryEventsApplied;
-  const eventCreatedTime = currentHistoryEvent?.created;
+  const sliderValue = numHistoryEventsApplied ?? 0;
+  const eventCreatedTime = currentHistoryEntry?.created;
   const playbackDisabled = numHistoryEventsApplied === undefined || sliderValue === history.length;
 
-  const handlePlayPauseToggle = (playing?: boolean) => {
+  const handlePlayPauseToggle = useCallback((playing?: boolean) => {
     const playStatus = playing !== undefined ? playing : !sliderPlaying;
-    Logger.logHistoryEvent({
-      documentId: treeManager.mainDocument?.key || '',
-      historyEventId: currentHistoryEvent?.id,
-      historyLength: history.length,
-      historyIndex: sliderValue,
-      action: playStatus ? "playStart": "playStop",
-    });
+    logCurrentHistoryEvent(treeManager, playStatus ? "playStart" : "playStop");
     setSliderPlaying(playStatus);
-  };
+  }, [sliderPlaying, treeManager]);
```

Comment:
In addition to refactoring the logging code I also took the opportunity to fix a React useEffects dependence warning.